src/plugins/kibana_react/public/empty_state_page/empty_state_page.tsx

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { EuiPageTemplate } from '@elastic/eui';
+import React from 'react';
+import { KibanaPageTemplateProps, NO_DATA_PAGE_TEMPLATE_PROPS, NoDataPage } from '../page_template';
+
+interface Props {
+  classes: string;
+}
+
+type EmptyPageProps = Props & KibanaPageTemplateProps;
+
+export const EmptyStatePage = (props: EmptyPageProps) => {
```

Comment:
Also, if we do keep this separate, I'd like to continue with the `No data` naming scheme as to not confuse this with any other type of empty page.
```suggestion
export const NoDataTemplate = (props: NoDataTemplateProps) => {
```